myceliumd/src/main.rs

```diff
@@ -268,10 +273,83 @@ pub struct NodeArguments {
     firewall_mark: Option<u32>,
 }
 
+#[derive(Debug, Deserialize, Default)]
+struct MyceliumConfig {
+    #[serde(deserialize_with = "deserialize_optional_endpoint_str_from_toml")]
+    peers: Option<Vec<Endpoint>>,
+    tcp_listen_port: Option<u16>,
+    quic_listen_port: Option<u16>,
+    no_tun: Option<bool>,
+    tun_name: Option<String>,
+    disable_peer_discovery: Option<bool>,
+    peer_discovery_port: Option<u16>,
+    api_addr: Option<SocketAddr>,
+    metrics_api_address: Option<SocketAddr>,
+    firewall_mark: Option<u32>,
+}
+
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
     let cli = Cli::parse();
 
+    // Init default configuration
+    let mut mycelium_config = MyceliumConfig::default();
+
+    // Load configuration file
+    if let Some(config_file_path) = &cli.config_file {
+        if Path::new(config_file_path).exists() {
+            let config = config::Config::builder()
+                .add_source(config::File::new(
+                    config_file_path.to_str().unwrap(),
+                    config::FileFormat::Toml,
+                ))
+                .build()?;
+
+            mycelium_config = config.try_deserialize()?;
+        } else {
+            warn!(
+                "Specified config file {:?} was not found.",
+                config_file_path
+            );
```

Comment:
I think it would be better to error here and exit if the user explicitly provides a config file and it is not found. If the user explicitly provides a file, and said file turns out not to be there, it probably means the user did something wrong (typo, put file in wrong place, etc)